hive-event-listeners/apiary-gluesync-listener/src/main/java/com/expediagroup/apiary/extensions/gluesync/cli/GlueSyncCli.java

```diff
@@ -0,0 +1,221 @@
+    Configuration config = new Configuration();
+    this.apiaryGlueSync = new ApiaryGlueSync(config, true);
+    this.isIcebergTablePredicate = new IsIcebergTablePredicate();
+    String gluePrefix = System.getenv("GLUE_PREFIX");
+    this.gluePartitionService = new GluePartitionService(glueClient, gluePrefix);
+    this.glueDatabaseService = new GlueDatabaseService(glueClient, gluePrefix);
+  }
+
+  // Constructor for testing - allows injection of dependencies
+  public GlueSyncCli(ThriftHiveClientFactory thriftHiveClientFactory,
+      ThriftHiveClient thriftHiveClient,
+      IMetaStoreClient metastoreClient,
+      ApiaryGlueSync apiaryGlueSync,
+      GlueTableService glueTableService,
+      GluePartitionService gluePartitionService,
+      IsIcebergTablePredicate isIcebergTablePredicate,
+      GlueDatabaseService glueDatabaseService) {
+    this.thriftHiveClientFactory = thriftHiveClientFactory;
+    this.thriftHiveClient = thriftHiveClient;
+    this.metastoreClient = metastoreClient;
+    this.apiaryGlueSync = apiaryGlueSync;
+    this.isIcebergTablePredicate = isIcebergTablePredicate;
+    this.gluePartitionService = gluePartitionService;
+    this.glueDatabaseService = glueDatabaseService;
+  }
+
+  public void syncAll(CommandLine cmd) {
+    logger.debug("Starting GlueSync operation");
+
+    String dbRegex = cmd.getOptionValue("database-name-regex");
+    String tableRegex = cmd.getOptionValue("table-name-regex");
+    boolean verbose = cmd.hasOption("verbose");
+
+    logger.debug("Sync parameters: dbRegex={}, tableRegex={}, verbose={}", dbRegex, tableRegex, verbose);
+
+    // Default to false if not provided
+    boolean continueOnError = cmd.hasOption("continueOnError");
+    // boolean deleteGlueTables = cmd.hasOption("delete-glue-tables");
+    boolean deleteGluePartitions = !cmd.hasOption("keep-glue-partitions");
+
+    logger.debug("Additional parameters: continueOnError={}, deleteGluePartitions={}", continueOnError,
+        deleteGluePartitions);
+
+    boolean hadError = false;
+    try {
+      for (String dbName : metastoreClient.getAllDatabases()) {
+        if (dbName.matches(dbRegex)) {
+          logger.debug("Processing database: {}", dbName);
+          try {
+            for (String tableName : metastoreClient.getAllTables(dbName)) {
+              if (tableName.matches(tableRegex)) {
+                try {
+                  logger.info("Syncing table: {} in database: {}", tableName, dbName);
+                  syncTable(dbName, tableName, deleteGluePartitions, verbose);
+                } catch (Exception e) {
+                  hadError = true;
+                  logger.error("Error syncing table: {} in database: {}: {}", tableName, dbName, e.getMessage());
+                  if (!continueOnError) {
+                    throw new RuntimeException("Error during sync operation", e);
+                  }
+                }
+              }
+            }
+          } catch (org.apache.thrift.TException e) {
```

Comment:
Why do you need these many catches? I suppose you need just 1, and maybe another if you want to make sure to continue even if you cannot list tables for 1 db

Reply:
I refactored that quite a bit and took out most of the exception handling, just one on the outside and one on the inside (around table) now. It will only continue if a single table sync fails otherwise it will just throw exceptions all the way out.